src/server/game/AI/PlayerAI/PlayerAI.cpp

```diff
@@ -705,6 +705,29 @@ void PlayerAI::DoAutoAttackIfReady()
         DoMeleeAttackIfReady();
 }
 
+void PlayerAI::CancelAllShapeshifts()
+{
+    std::list<AuraEffect*> const& shapeshiftAuras = me->GetAuraEffectsByType(SPELL_AURA_MOD_SHAPESHIFT);
+    std::set<Aura*> removableShapeshifts;
+    for (AuraEffect* auraEff : shapeshiftAuras)
+    {
+        Aura* aura = auraEff->GetBase();
+        if (!aura)
+            continue;
+        SpellInfo const* auraInfo = aura->GetSpellInfo();
+        if (!auraInfo)
+            continue;
+        if (auraInfo->HasAttribute(SPELL_ATTR0_CANT_CANCEL))
```

Comment:
I do not agree with this whole method, you should use `RemoveAurasByType(SPELL_AURA_MOD_SHAPESHIFT);`
Look at https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L1655,
when player use an shapeshift, this line is called, nothing wrong here.

Side note, noone spell with SPELL_AURA_MOD_SHAPESHIFT has SPELL_ATTR0_CANT_CANCEL or IsPassive or IsPositive, everything seems redundant